pkg/apis/core/validation/validation.go

```diff
@@ -7080,6 +7082,9 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 		if err := ValidateProcMountType(fldPath.Child("procMount"), *sc.ProcMount); err != nil {
 			allErrs = append(allErrs, err)
 		}
+		if !inUserNS && sc.ProcMount != nil && *sc.ProcMount == core.UnmaskedProcMount {
```

Comment:
This is tightening validation. I'm not sure what the implications of that are for an alpha feature though. But you probably need an api reviewer to take a look (tag it with `/label api-review`). You probably need to plumb through an option to skip this validation if the object is already in an invalid state.

Reply:
can you explain further? I belive this field is stripped when the feature gate isn't enabled, so I'm not sure the condition we would add the pod validation option or not

Reply:
Changing validation while in alpha is fair game... I'm not sure I would add the complexity of ratcheting for it.

A release note is probably sufficient, but the validation change likely means we'll wait to move this to beta until 1.31